lib/templatesimplifier.cpp

```diff
@@ -1763,7 +1763,7 @@ void TemplateSimplifier::expandTemplate(
                     }
                 } else if (copy) {
                     bool added = false;
-                    if (tok5->isName()) {
+                    if (tok5->isName() && !Token::Match(tok5, "class|typename")) {
```

Comment:
What about `struct`?

Reply:
That would be a syntax error:

```
g++ test.cpp
test.cpp:1:17: error: expected identifier before ‘>’ token
 template <struct> class A { void f(); };
                 ^
test.cpp:2:6: error: type/value mismatch at argument 1 in template parameter list for ‘template<<typeprefixerror><anonymous> > class A’
 A<int> a;
      ^
test.cpp:2:6: note:   expected a constant of type ‘<type error>’, got ‘int’

```

Reply:
The same problem exists for struct:
`template <struct> struct A { A(); };` turns into `int A<int>@2 { A<int>@2 ( ) ; } ;`
I guess checking for `struct` will make sure another syntax error is not added.